Production checklist v1 (#691)
seqeralabs/docs: platform-cloud/docs/getting-started/production-checklist.md

## Users and roles

Once an organization is created, the user who created the organization is the default owner of that organization. Additional users can be assigned as organization owners. Owners have full read/write access to modify members, teams, collaborators, and settings within an organization.

- A member is a user who is internal to the organization.
- Members have an organization role and can operate in one or more organization workspaces.
- In each workspace, members have a participant role that defines the permissions granted to them within that workspace.
- Members can create their own workspaces within an organization and be part of a team.

All users can be assigned roles which granulate the type of access and permissions they have to resources within Platform. 

It’s a good idea to map out the expected users and their roles so you can ensure your plans are scalable. Read [our documentation](https://docs.seqera.io/platform-enterprise/25.1/orgs-and-teams/roles) for more information.

## Infrastructure

Infrastructural requirements differ widely based on the workload you’re expecting. 

> To begin with, build out a proof of concept using the below recommendations, to create a baseline of your capacity requirements. Once you’re ready to move to production, take into account the increased workload you’d expect. Here are some starting points for estimating compute and database requirements.

Reviewer: Do we have any examples we can provide as a good template for getting started?

Author: I've added @gavinelder as a reviewer and he may be able to provide more guidance.

- Amazon Machine Image (AMI): Amazon Linux 2023 Optimized
- Instance type: `c5a.2xlarge` with 8 CPUs and 16 GB RAM
- A MySQL8 Community DB instance with minimum 2 vCPUs, 8 GB memory, and 30 GB SSD storage

This sizing recommendation is a basic starting point. Your requirements may vary significantly based on the number of pipelines, and number of concurrent processes, you have in mind. Consult the [AWS Autoscaling documentation](https://aws.amazon.com/autoscaling/) information about resource management in AWS.

### Azure

When deploying Seqera Enterprise using Azure we recommend starting with: 

- Azure Linux VM with default values
- At least 2 CPUS and 8 GB RAM
- Ubuntu Server 22.04 LTS - Gen2 image
- A MySQL8 Community DB instance with minimum 2 vCPUs, 8 GB memory, and 30 GB SSD storage

These autoscale for pipeline runs, but the sizing recommendation will be based on the workload and can vary significantly based on the number of pipelines, and number of concurrent processes, you have in mind. Consult the [Azure autoscaling documentation](https://learn.microsoft.com/en-us/azure/azure-monitor/autoscale/autoscale-get-started) for information about scaling in Azure.

> ## Spot instance retry strategy

Reviewer: Would we provide an example here or link to one?

Author: The tutorial we link to does cover the retry strategy but you make a good point. Maybe this would be better as a list with links to sections within the tutorial.

One way to mitigate issues is having a robust retry and resume strategy. Refer to [our retry strategy tutorial](https://docs.seqera.io/platform-cloud/tutorials/retry-strategy) about this for more information.  

## Cost management and alerts

Managing your compute spend upfront is a critical part of your production deployment. Here are some best practices that we recommend:

- Utilize AWS Batch job tagging. This is facilitated by Nextflow's configuration and can be crucial in tracing costs back to specific pipelines. They can include dynamic variables and can be a valuable tool for helping diagnose and identify unexpected fees. This is especially helpful if you’re using Nextflow outside of Seqera Platform.
- Note that CloudWatch fees are not included in the cost estimate for runs.

## Seqera Platform limitations

> When cancelling large runs, make sure to check that all jobs were killed in your compute environment (ZOMBIE JOBS). Large runs sometimes leak jobs because Nextflow is killed before it can cancel all of them, which can lead to significant cost overruns.

Reviewer: How can I do this as a user? How do I know what a  zombie job is?

## Users and roles

Once an organization is created, the user who created the organization is the default owner of that organization. Additional users can be assigned as organization owners. Owners have full read/write access to modify members, teams, collaborators, and settings within an organization.

- A member is a user who is internal to the organization.
- Members have an organization role and can operate in one or more organization workspaces.
- In each workspace, members have a participant role that defines the permissions granted to them within that workspace.
- Members can create their own workspaces within an organization and be part of a team.

All users can be assigned roles which granulate the type of access and permissions they have to resources within Platform. 

It’s a good idea to map out the expected users and their roles so you can ensure your plans are scalable. Read [our documentation](https://docs.seqera.io/platform-enterprise/25.1/orgs-and-teams/roles) for more information.

## Infrastructure

> Infrastructural requirements differ widely based on the workload you’re expecting. The [Seqera Enterprise architecture overview](https://docs.seqera.io/platform-enterprise/latest/enterprise/overview) provides an outline of the components of Platform.

Reviewer 2: ```suggestion
Infrastructural requirements vary widely depending on the workload you expect. See [Enterprise installation](https://docs.seqera.io/platform-enterprise/latest/enterprise/overview) for an outline of Platform components.
```

### AWS

When deploying Seqera Enterprise using AWS we recommend starting with: 

- Amazon Machine Image (AMI): Amazon Linux 2023 Optimized
- Instance type: `c5a.2xlarge` with 8 CPUs and 16 GB RAM
- A MySQL8 Community DB instance with minimum 2 vCPUs, 8 GB memory, and 30 GB SSD storage

This sizing recommendation is a basic starting point. Your requirements may vary significantly based on the number of pipelines, and number of concurrent processes, you have in mind. Consult the [AWS Autoscaling documentation](https://aws.amazon.com/autoscaling/) information about resource management in AWS.

### Azure

When deploying Seqera Enterprise using Azure we recommend starting with: 

- Azure Linux VM with default values
- At least 2 CPUS and 8 GB RAM
- Ubuntu Server 22.04 LTS - Gen2 image
- A MySQL8 Community DB instance with minimum 2 vCPUs, 8 GB memory, and 30 GB SSD storage

> These autoscale for pipeline runs, but the sizing recommendation will be based on the workload and can vary significantly based on the number of pipelines, and number of concurrent processes, you have in mind. Consult the [Azure autoscaling documentation](https://learn.microsoft.com/en-us/azure/azure-monitor/autoscale/autoscale-get-started) for information about scaling in Azure.

Reviewer 2: ```suggestion
These autoscale for pipeline runs, but the sizing recommendation will be based on the workload and can vary significantly based on the number of pipelines and concurrent processes you anticipate. See [Azure autoscaling documentation](https://learn.microsoft.com/en-us/azure/azure-monitor/autoscale/autoscale-get-started) for information about scaling in Azure.
```

- [Handle retries in Nextflow by setting `errorStrategy` and `maxRetries`](https://docs.seqera.io/platform-cloud/tutorials/retry-strategy#handle-retries-in-nextflow-by-setting-errorstrategy-and-maxretries)
- [Handle retries in AWS by setting `aws.batch.maxSpotAttempts`](https://docs.seqera.io/platform-cloud/tutorials/retry-strategy#handle-retries-in-aws-by-setting-awsbatchmaxspotattempts)
- [Implement Spot to On-Demand fallback logic](https://docs.seqera.io/platform-cloud/tutorials/retry-strategy#implement-spot-to-on-demand-fallback-logic)

Refer to [our retry strategy tutorial](https://docs.seqera.io/platform-cloud/tutorials/retry-strategy) for more information.

## Cost management and alerts

Managing your compute spend upfront is a critical part of your production deployment. Here are some best practices that we recommend:

- Utilize AWS Batch job tagging. This is facilitated by Nextflow's configuration and can be crucial in tracing costs back to specific pipelines. They can include dynamic variables and can be a valuable tool for helping diagnose and identify unexpected fees. This is especially helpful if you’re using Nextflow outside of Seqera Platform.

> - Note that CloudWatch fees are not included in the cost estimate for runs.

Reviewer 2: ```suggestion
- Note that CloudWatch fees are not included in your run cost estimates.
```
